solutions/CSES Sliding Cost.mdx

```diff
@@ -0,0 +1,98 @@
+---
+id: cses-1077
+source: CSES
+title: Sliding Cost
+author: Isaac Noel
+---
+
+**Time Complexity:** $\mathcal{O}(N\log K)$
+
+## Abstract
+
+Finding the difference between sums of the upper $K/2$ elements and lower $K/2$ with two multisets.
+
+<Warning>
+
+The implementation of this solution assumes familarity with [Sliding Median](/gold/sliding#problem-https://cses.fi/problemset/task/1076) as their implementations are near identical.
+
+</Warning>
+
+## Solution
+
+### The cost of a window
+
+It can be shown that it is optimal to change the values of the window to the median (left as an exercise :D). Once we find the median, we must find the sum of the distance of all elements to the median. Rather than finding the absolute value of every individual element's distance, which is far too slow, we can split the window into two groups and use $\mathcal{O}(\log K)$ insertions/deletions and a calculation for the cost that runs in constant time. The smallest $K/2$ elements in the window will be in the lower group while the largest $K/2$ elements in the window will be in the upper group.
```

Comment:
> we must find the sum of the distance of all elements

distances

Reply:
> Rather than finding the absolute value of every individual element's distance, which is far too slow, we can split the window into two groups and use $\mathcal{O}(\log K)$ insertions/deletions and a calculation for the cost that runs in constant time.

What about -

> Summing the distances for each element individually is too slow. Instead, we'll split the elements in the window into two groups and calculate the cost as described below.